fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1224,118 +1225,118 @@ ACTOR static Future<Void> transactionStarter(
 	}
 }
 
-ACTOR static Future<Void> readRequestServer(
-	MasterProxyInterface proxy,
-	ProxyCommitData* commitData
-	)
-{
+ACTOR static Future<Void> readRequestServer( MasterProxyInterface proxy, ProxyCommitData* commitData ) {
 	// Implement read-only parts of the proxy interface
-
 	// We can't respond to these requests until we have valid txnStateStore
 	wait(commitData->validState.getFuture());
 
 	TraceEvent("ProxyReadyForReads", proxy.id());
 
 	loop {
-		choose{
```

Comment:
Just out of curiosity, why did you ditch the `choose-when` here?

Reply:
I am asking cause I just found in Document Layer's code that a special combination of `loop-choose-when` can cause priority inversion.

Reply:
It is because of the yield which is in the loop. When the proxy starts this actor it may already have a huge backlog of KeyServerLocationsRequests, so we want to yield to prevent a slow task. But this yield allows more requests to be queued up, so in saturation we can enter a state where we always have a backlog of KeyServerLocationsRequests. In this mode, the proxy will then never process any RejoinRequests because it is lower in the choose-when block.

Reply:
Yes this is consistent with the bug I saw. Basically if the first when is for a lower priority task and its block contains a wait(), it's likely to introduce priority inversion under certain cases, not necessarily to be saturation. Just a bad luck can get the loop into this state.

It goes beyond the scope of this PR but I think we need to warn users in Flow compiler if it sees this patter (a wait in the first when statement.)